lcls2_producers/smd_producer.py

```diff
@@ -123,218 +130,245 @@ def define_dets(run, det_list):
-                    action='store_true',
-                    default=False)
-parser.add_argument('--nohsd',
-                    help="dont save HSD data",
-                    action='store_true',
-                    default=False)
-parser.add_argument('--nosum',
-                    help="dont save sums",
-                    action='store_true',
-                    default=False)
-parser.add_argument('--noarch',
-                    help="dont use archiver data",
-                    action='store_true',
-                    default=False)
+parser.add_argument(
+    "--run", help="run", type=str, default=os.environ.get("RUN_NUM", "")
+)
+parser.add_argument(
+    "--experiment",
+    help="experiment name",
+    type=str,
+    default=os.environ.get("EXPERIMENT", ""),
+)
+parser.add_argument("--stn", help="hutch station", type=int, default=0)
+parser.add_argument("--nevents", help="number of events", type=int, default=0)
+parser.add_argument(
+    "--directory", help="directory for output files (def <exp>/hdf5/smalldata)"
+)
+parser.add_argument("--gather_interval", help="gather interval", type=int, default=100)
+parser.add_argument(
+    "--norecorder", help="ignore recorder streams", action="store_true", default=False
+)
+parser.add_argument("--url", default="https://pswww.slac.stanford.edu/ws-auth/lgbk/")
+parser.add_argument(
+    "--epicsAll", help="store all epics PVs", action="store_true", default=False
+)
+parser.add_argument(
+    "--full",
+    help="store all data (please think before usig this)",
+    action="store_true",
+    default=False,
+)
+parser.add_argument(
+    "--default", help="store only minimal data", action="store_true", default=False
+)
+parser.add_argument(
+    "--image",
+    help="save everything as image (use with care)",
+    action="store_true",
+    default=False,
+)
+parser.add_argument(
+    "--tiff",
+    help="save all images also as single tiff (use with even more care)",
+    action="store_true",
+    default=False,
+)
+parser.add_argument(
+    "--postRuntable",
+    help="postTrigger for seconday jobs",
+    action="store_true",
+    default=False,
+)
+parser.add_argument(
+    "--wait", help="wait for a file to appear", action="store_true", default=False
+)
+parser.add_argument(
+    "--rawFim", help="save raw Fim data", action="store_true", default=False
+)
+parser.add_argument(
+    "--nohsd", help="dont save HSD data", action="store_true", default=False
+)
+parser.add_argument(
+    "--nosum", help="dont save sums", action="store_true", default=False
+)
+parser.add_argument(
+    "--noarch", help="dont use archiver data", action="store_true", default=False
+)
+parser.add_argument(
+    "--psplot_live_mode",
+    help="Run as a server for psplot live mode, i.e. no h5 file being written.",
+    action="store_true",
+    default=False,
+)
+parser.add_argument(
+    "--intg_delta_t",
+    help="Offset for the integrating detector batch.",
+    type=int,
+    default=0,
+)
```

Comment:
New arguments.

Note that for psplot_live_mode, we do not write ti h5 anymore and force a single SRV core.